lndr-backend/src/Lndr/Server.hs

```diff
@@ -228,19 +230,32 @@ verifySettlementsWithTxHash :: LndrHandler ()
 verifySettlementsWithTxHash = do
     (ServerState pool configTVar _) <- ask
     config <- liftIO $ readTVarIO configTVar
-    creditHashes <- liftIO $ withResource pool Db.settlementCreditsToVerify
-    mapM_ verifyIndividualRecord creditHashes
+    txHashes <- liftIO $ withResource pool Db.txHashesToVerify
+    creditsToVerify <- mapM (\txHash -> liftIO $ withResource pool $ Db.lookupCreditsByTxHash txHash) txHashes
+    mapM_ (\txGroupList -> verifyRecords txGroupList) creditsToVerify
 
 
-verifyIndividualRecord :: TransactionHash -> LndrHandler ()
-verifyIndividualRecord creditHash = do
+verifyRecords :: [BilateralCreditRecord] -> LndrHandler ()
+verifyRecords records = do
     (ServerState pool configTVar loggerSet) <- ask
-    recordM <- liftIO $ withResource pool $ Db.lookupCreditByHash creditHash
-    let recordNotFound = throwError $
-            err404 { errBody = "Credit hash does not refer to pending bilateral settlement record" }
-    bilateralCreditRecord <- maybe recordNotFound pure recordM
-    verifySettlementPayment bilateralCreditRecord
-    liftIO $ withResource pool $ Db.verifyCreditByHash creditHash
-    web3Result <- finalizeTransaction configTVar bilateralCreditRecord
-                        `catchError` (pure . T.pack . show)
+    
+    -- this should only take the creditor, debtor, credit hash, and settlement amount
+    let initialTxHash = txHash $ head records
+        initialRecord = creditRecord $ head records
+        initialSettlement = SettlementVerification (creditor initialRecord) (debtor initialRecord) initialTxHash (fromMaybe 0 $ settlementAmount initialRecord)
+        settlementRecord = foldr generateCompositeSettlement initialSettlement (tail records)
+    verifySettlementPayment settlementRecord
+    mapM_ (\(BilateralCreditRecord creditRecord _ _ _) -> liftIO $ withResource pool $ Db.verifyCreditByHash $ hash creditRecord) records
```

Comment:
use : `mapM_ (liftIO . withResource pool . Db.verifyCreditByHash . hash . creditRecord) records`

to get at the `creditRecord` in the `BilateralCreditRecord` type, just use the record field accessor function. read this: https://en.wikibooks.org/wiki/Haskell/More_on_datatypes#Named_Fields_(Record_Syntax)